pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java

```diff
@@ -882,6 +882,7 @@ public Void call() throws Exception {
               e.getSQLState()
           );
           cancels++;
+          break;
```

Comment:
@jajalvipul , can you please clarify why the test takes so long to finish?

Apparently it was supposed to be a fast test, so I'm curious what are the threads doing if it test takes 100 seconds.

Reply:
@vlsi 

Yes, you are right that cancellation work "immediately" but seems to be in Linux environment.
The cancellation does work in Windows platform but it is not as fast as compared to Linux platform.

In Windows platforms, while running the JDBC tests, it has been observed the cancellation work but it starts to slow down with every iteration in the for loop present in the method `StatementTest#testCloseInProgressStatement`. And eventually it exceeds the JUnit timeout (10 sec) set for this method.

Most of the times, the JUnit timeout was exceeding, terminating the test method execution. This was resulting in hang issue at the `TestUtil.dropTable(con, "test_lock");` in `tearDown()`.

Couple of times, following exception 'IllegalStateException' was also encountered, resulting in hang issue as mentioned above.

```java   
         public Void call() throws Exception {
            long start = System.currentTimeMillis();
            while (st.getWarnings() == null) {
              long dt = System.currentTimeMillis() - start;
              if (dt > 10000) {
                throw new IllegalStateException("Expected to receive a notice within 10 seconds");
              }
            }
            st.close();
            return null;
          }

```